cmd/agent/subcommands/run/command.go

```diff
@@ -503,18 +490,41 @@ func startAgent(
 		}
 	}
 
+	var logsAgent *logs.Agent
+
 	// start logs-agent.  This must happen after AutoConfig is set up (via common.LoadComponents)
 	if pkgconfig.Datadog.GetBool("logs_enabled") || pkgconfig.Datadog.GetBool("log_enabled") {
 		if pkgconfig.Datadog.GetBool("log_enabled") {
 			log.Warn(`"log_enabled" is deprecated, use "logs_enabled" instead`)
 		}
-		if _, err := logs.Start(common.AC); err != nil {
+		var err error
+		if logsAgent, err = logs.Start(common.AC); err != nil {
 			log.Error("Could not start logs-agent: ", err)
 		}
 	} else {
 		log.Info("logs-agent disabled")
 	}
 
+	// Start OTLP intake
+	otlpEnabled := otlp.IsEnabled(pkgconfig.Datadog)
+	inventories.SetAgentMetadata(inventories.AgentOTLPEnabled, otlpEnabled)
+	if otlpEnabled {
+		if logsAgent == nil {
```

Comment:
This will break all existing users who uses OTLP but does not enable log agent - logsAgent is only needed when otlp_config::logs is present